geomesa-gs-monitor-elasticsearch/src/main/scala/org/geomesa/gs/monitor/elastic/ElasticRequestDataListener.scala

```diff
@@ -8,137 +8,165 @@
 
 package org.geomesa.gs.monitor.elastic
 
-import java.lang.reflect.Type
-import java.net.{MalformedURLException, URL}
-import java.util.Date
-
-import com.google.gson._
+import com.typesafe.config.{Config, ConfigFactory}
 import com.typesafe.scalalogging.LazyLogging
 import org.apache.http.HttpHost
-import org.elasticsearch.action.ActionListener
-import org.elasticsearch.client.{RequestOptions, RestClient, RestHighLevelClient}
-import org.elasticsearch.action.index.{IndexRequest, IndexResponse}
-import org.elasticsearch.common.xcontent.XContentType
-import org.elasticsearch.action.DocWriteResponse
-import org.geoserver.monitor.{RequestData, RequestDataListener}
-import org.opengis.geometry.BoundingBox
-
-import scala.collection.mutable.ArrayBuffer
-
-
+import org.apache.http.auth.{AuthScope, UsernamePasswordCredentials}
+import org.apache.http.impl.client.BasicCredentialsProvider
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder
+import org.elasticsearch.client.RestClientBuilder.HttpClientConfigCallback
+import org.elasticsearch.client.{Request, RestClient}
+import org.geomesa.gs.monitor.elastic.ExtendedRequestData.TIMEOUT_KEY
+import org.geoserver.monitor
+import org.geoserver.monitor.RequestData.Status
+import org.geoserver.monitor.RequestDataListener
+
+import java.util.concurrent.{Executors, LinkedBlockingQueue, TimeUnit}
+import scala.collection.JavaConverters._
+import scala.concurrent.TimeoutException
 
 class ElasticRequestDataListener extends RequestDataListener with LazyLogging {
-  import ElasticRequestDataListener.gson
-  val envVars = sys.env.getOrElse("ELASTICSEARCH_HOST", "unset").split(',')
-  if(envVars(0) == "unset"){
-    logger.error("Environment variable ELASTICSEARCH_HOST is not set")
-    throw new Exception("Environment variable ELASTICSEARCH_HOST is not set. Stopping build.")
-  }
-  //initialized variables
-  var host = ""
-  var port = 0
-  var protocol = ""
-  val hostList = new ArrayBuffer[HttpHost]
-
-  for (u <- envVars) {
-    try {
-      val url = new URL(u.trim())
-      host = url.getHost
-      port = url.getPort
-      protocol = url.getProtocol
-      hostList.append(new HttpHost(host, port, protocol))
-    }
-    catch{
-      case e: MalformedURLException => logger.error("Invalid URL " + u + ". Could not convert from string to URL. Trying any additional URLs")
+
+  import org.geomesa.gs.monitor.elastic.ElasticRequestDataListener._
+
+  private val config = ConfigFactory.load.getConfig(GEOSERVER_MONITOR_ELASTICSEARCH_KEY)
```

Comment:
not needed for this PR, but it might make it simpler to configure in dockers if we provided a default reference.conf that delegated to env vars for each prop, i.e. `geomesa.geoserver.monitor.elasticsearch.host = ${?MONITOR_ES_HOST}`

Reply:
So I would just put the reference.conf in src/main/resources? And the application.conf would override the env vars or the other way around?

Reply:
yeah i think any application.conf would override it. if you want to configure it via application.conf, you'd just use the application.conf in the same way you're currently doing. but instead, you could also just set the env vars and those would get picked up without having to modify the docker or mount in an application.conf file. the `${?foo}` syntax should remove the entire line if the env var isn't set, so i don't think it would affect anything in that case. if you set both env vars and had an application.conf, i think the application.conf would take precedence, but generally you'd have either one or the other. would probably require some testing to make sure it works the way i think...

Reply:
Seems to work as you said (using environment variables for a list config is a pain though)

Reply:
that's a neat trick for the list config!